### PR TITLE
task/alerts/alert_signal

### PR DESCRIPTION
**Pull request by BarAgam**
Notifying user when a new alerts is received.
App's toolbar is animating from its original color to red (repeatedly) and it's (the toolbar's) title is changing to alert.
The toolbar status is being restored after clicking on it.

### CODE REVIEW

File: app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    @Inject
+    BubbleAlertsViewModelFactory mBubbleAlertsViewModelFactory;
+
+    private final CharSequence mToolbarOriginalTitle;
+    private final BubbleAlertsViewModel mBubbleAlertsViewModel;
+    private final Toolbar mToolbar;
+
+
+    public BubbleAlertsSubcomponent(Activity activity) {
+        mToolbar = (Toolbar) activity.findViewById(R.id.main_toolbar);
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java:30]: ButterKnife can be used here

File: app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                    new ArgbEvaluator(), from, to);
+
+            mColorAnimation.setDuration(ANIMATION_DURATION_MS);
+            mColorAnimation.setStartDelay(ANIMATION_START_DELAY_MS);
+            mColorAnimation.setRepeatCount(ValueAnimator.INFINITE);
+            mColorAnimation.setRepeatMode(ValueAnimator.REVERSE);
+        }
+    }
+
+    private class MyTitleSetter implements BubbleAlertsViewModel.TitleSetter {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java]: Change the name of the title setter

File: app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+    private MyToolbarAnimator createActionbarAnimator(Activity activity) {
+        return new MyToolbarAnimator(mToolbar, activity.getResources());
+    }
+
+    private void onToolbarClicked() {
+        mBubbleAlertsViewModel.onToolbarClick();
+    }
+
+    private class MyToolbarAnimator implements BubbleAlertsViewModel.ToolbarAnimator {
```
Comment (teamagam3) [inline app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java]: Changed the name of the class

File: data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/AlertsDataRepository.java [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
     @Inject
     public AlertsDataRepository() {
-        mReplayRepository = ReplayRepository.createReplayAll();
+        mReplayRepository = ReplayRepository.createReplayCount(1);
```
Comment (teamagam3) [inline data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/AlertsDataRepository.java]: I understand why you changed this, but what if someone needs to get all of the alerts?

File: data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/InformedAlertsDataRepository.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import javax.inject.Singleton;
+
+@Singleton
+public class InformedAlertsDataRepository implements InformedAlertsRepository {
+
+    private Date mLatestInformedDate;
+
+    @Inject
+    public InformedAlertsDataRepository() {
+        mLatestInformedDate = new Date(0);
```
Comment (teamagam3) [inline data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/InformedAlertsDataRepository.java:17]: Is there a Date.minValue() or something similar?

File: data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java [hunk 6]
```diff
@@ -3,7 +3,7 @@
 import android.graphics.Bitmap;
 
 public class Constants {
-    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev2.herokuapp.com";
+    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev1.herokuapp.com";
```
Comment (teamagam3) [inline data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java:6]: Revert changes

File: domain/src/main/java/com/teamagam/gimelgimel/domain/alerts/UpdateLatestInformedAlertTimeInteractor.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+import com.teamagam.gimelgimel.domain.base.interactors.BaseDataInteractor;
+import com.teamagam.gimelgimel.domain.base.interactors.DataSubscriptionRequest;
+
+import java.util.Collections;
+
+import rx.Observable;
+
+@AutoFactory
+public class UpdateLatestInformedAlertTimeInteractor extends BaseDataInteractor {
```
Comment (teamagam3) [inline domain/src/main/java/com/teamagam/gimelgimel/domain/alerts/UpdateLatestInformedAlertTimeInteractor.java:16]: Don't we need a simpler class for this functionality?
It's not real "subsription request", we just want to update the InformedAlertsRepository.

Maybe I just don't understand the real meaning of this kind of class. Isn't it  just for long streams like messages streams?

**Review by teamagam3 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java (hunk 1 above)
Comment (BarAgam) [reply app/src/main/java/com/teamagam/gimelgimel/app/Alerts/view/BubbleAlertsSubcomponent.java:30]: Unfortunately I couldn't make it happen here.

File: data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/AlertsDataRepository.java (hunk 4 above)
Comment (BarAgam) [reply data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/AlertsDataRepository.java]: That's right.

File: data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/InformedAlertsDataRepository.java (hunk 5 above)
Comment (BarAgam) [reply data/src/main/java/com/teamagam/gimelgimel/data/alerts/repository/InformedAlertsDataRepository.java:17]: Nope.

File: domain/src/main/java/com/teamagam/gimelgimel/domain/alerts/UpdateLatestInformedAlertTimeInteractor.java (hunk 7 above)
Comment (BarAgam) [reply domain/src/main/java/com/teamagam/gimelgimel/domain/alerts/UpdateLatestInformedAlertTimeInteractor.java:16]: Basically we're using interactors for everything kind of action we want to make on our domain.
It's a good for several reasons, like uniformly manipulating domain from app and allows for logic to be implemented independently for each "use-case" (=interactors).

**Review by teamagam3 — APPROVED** [review]

**Pull request merged**